### PR TITLE
Add a simple example of backup and restore

### PR DESCRIPTION
**Pull request by YiteGu**
Add a simple example of backup and restore

Signed-off-by: YiteGu <ess_gyt@qq.com>

### CODE REVIEW

**Review by ajkr — APPROVED** [review]
LGTM, thanks

**Comment by YiteGu** [comment]
@ajkr thanks for you code review :)

**Comment by YiteGu** [comment]
hi @ajkr
CI check is done, please help merge this pr.

**Comment by YiteGu** [comment]
hi @pdillinger 
please help re-trigger Travis CI

File: examples/rocksdb_backup_restore_example.cc [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+  s = backup_engine->VerifyBackup(1);
+  assert(s.ok());
+
+  // Put key-value
+  db->Put(WriteOptions(), "key2", "value2");
+  assert(s.ok());
+
+  db->Close();
+  db = nullptr;
```
Comment (ajkr) [inline examples/rocksdb_backup_restore_example.cc:69]: Do you need to `delete db` before this?

Comment (YiteGu) [reply examples/rocksdb_backup_restore_example.cc:69]: hmm, I should free memory space of db. 
Should I add `delete db` between `db->Close` and `db = nullptr`?

Comment (ajkr) [reply examples/rocksdb_backup_restore_example.cc:69]: Yes

**Pull request closed without merging**